### PR TITLE
[MacOS] Fix stack-buffer-underflow

### PR DESCRIPTION
**Pull request by enen92**
## Description
Found while running address sanitiser in xcode (thanks to @pkerling)
Log here: https://paste.kodi.tv/tawewohiye.kodi

## Motivation and Context
Kodi won't start from xcode if asan is enabled. Can help debugging further issues.

## How Has This Been Tested?
Compile and run with asan enabled

## Types of change

- [x] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [x] All new and existing tests passed

### CODE REVIEW

**Review by Memphiz — APPROVED** [review]
What is with the other places where CGetActiveDisplayList is called? They look similar no?

Thx for the fix :)

File: xbmc/windowing/osx/WinSystemOSX.mm [hunk 1]
```diff
@@ -195,7 +195,11 @@ CGDirectDisplayID GetDisplayID(int screen_index)
 
   // Get the list of displays.
   CGGetActiveDisplayList(MAX_DISPLAYS, displayArray, &numDisplays);
-  return(displayArray[screen_index]);
+
+  if (screen_index >= 0 && screen_index <= numDisplays)
```
Comment (Memphiz) [inline xbmc/windowing/osx/WinSystemOSX.mm]: Ah wait - the check needs to be screen_index < numDisplays...

Comment (enen92) [reply xbmc/windowing/osx/WinSystemOSX.mm]: ups, you're right. Fixed.

**Comment by enen92** [comment]
@Memphiz looking at the code it looks like it can also happen in other places where `CGetActiveDisplayList` is called. As I was trying to figure out another bug this fix let me run kodi so it might happen that I haven't run into the other calls. Do you want me to do the same for the others as well?

**Comment by enen92** [comment]
Jenkins build this please

**Comment by MartijnKaijser** [comment]
@Memphiz any comments?

**Pull request merged**